libenv/unix_iface.c

```diff
@@ -760,38 +759,50 @@ static int aix_get_mac_addr(const char *device_name, uint8_t mac[6])
 }
 #endif /* _AIX */
 
+// <bahamat> And instead of "ipv6_addresses" I'd use 'inet6', or "AF_INET6"...something like that.  [13:45]
+// <bahamat> And I wouldn't have 'routes' or 'addresses' ever unqualified  [13:46]
+// <bahamat> In fact, I'd recomend {"inet": {"addresses": {}, "routes": {}}, "inet6": {"addresses": {}, "routes": {}}
+
 // TODO: perhaps rename and move these to json.c and ip_address.c?  Or even let JsonElements store IPAddress structs?
-static Buffer* ParsedIPAddressHex(const char *data, const bool with_port)
+static IPAddress* ParsedIPAddressHex(const char *data)
 {
-    Buffer *addr = NULL;
+    IPAddress *ip = NULL;
     Buffer *buffer = BufferNewFrom(data, strlen(data));
     if (NULL != buffer)
     {
-        IPAddress *ip = IPAddressNewHex(buffer);
-        if (NULL != ip)
-        {
-            addr = IPAddressGetAddress(ip);
-            int port = IPAddressGetPort(ip);
-            if (with_port)
-            {
-                BufferAppendF(addr, ":%d", port);
-            }
-            IPAddressDestroy(&ip);
-        }
+        ip = IPAddressNewHex(buffer);
         BufferDestroy(buffer);
     }
 
-    return addr;
+    return ip;
 }
 
-static void JsonRewriteParsedIPAddress(JsonElement* element, const char* raw_key, const char *new_key, const bool with_port)
+static void JsonRewriteParsedIPAddress(JsonElement* element, const char* raw_key, const char *new_key, const bool as_map)
 {
-    Buffer *addr = ParsedIPAddressHex(JsonObjectGetAsString(element, raw_key), with_port);
+    IPAddress *addr = ParsedIPAddressHex(JsonObjectGetAsString(element, raw_key));
     if (NULL != addr)
     {
-        JsonObjectAppendString(element, new_key, BufferData(addr));
-        JsonObjectRemoveKey(element, raw_key);
-        BufferDestroy(addr);
+        Buffer *buf = IPAddressGetAddress(addr);
+        if (NULL != buf)
+        {
+            if (as_map)
+            {
+                JsonElement *ip = JsonObjectCreate(2);
+                JsonObjectAppendString(ip, "address", BufferData(buf));
+                BufferPrintf(buf, "%d", IPAddressGetPort(addr));
+                JsonObjectAppendString(ip, "port", BufferData(buf));
+                JsonObjectAppendElement(element, new_key, ip);
+            }
+            else
+            {
+                JsonObjectAppendString(element, new_key, BufferData(buf));
+            }
+
+            BufferDestroy(buf);
+            JsonObjectRemoveKey(element, raw_key);
```

Comment:
Could `raw_key` and `new_key` theoretically be the same? If so I assume that `JsonObjectAppendElement` would in fact not append the element, but overwrite it, with it being a map and all (I haven't actually verified this though). Then you should not remove the element afterwards.